graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/aggregations/AggregationsFetcherSupport.java

```diff
@@ -0,0 +1,222 @@
+            putResultValue(columns, row, selectedField, Row::getByte);
+            break;
+        }
+      }
+    }
+    return columns;
+  }
+
+  private void putResultValue(
+      Map<String, Object> columns,
+      Row row,
+      SelectedField selectedField,
+      BiFunction<Row, String, Object> rowValueExtractor) {
+    String alias = selectedField.getAlias();
+    // put the returned value as alias
+    if (alias != null) {
+      columns.put(selectedField.getName(), rowValueExtractor.apply(row, alias));
+    }
+    // generate aggregation column name and get the value
+    else {
+      String columnName = generateAggregationColumnName(selectedField);
+      columns.put(selectedField.getName(), rowValueExtractor.apply(row, columnName));
+    }
+  }
+
+  // it returns C* generated aggregation column name
+  // for example, for the name = count and arguments = ['a']
+  // it will return: 'system.count(a)'
+  private String generateAggregationColumnName(SelectedField selectedField) {
+    Map<String, Object> arguments = selectedField.getArguments();
+    SupportedAggregationFunction functionName = getNameArgument(arguments);
+    List<String> args = getAndValidateArgs(arguments, functionName);
+    return String.format("system.%s(%s)", functionName.getName(), args.get(0));
+  }
+
+  private void addAggregation(
+      Map<String, Object> arguments,
+      QueryBuilder.QueryBuilder__20 queryBuilder,
+      BiFunction<QueryBuilder.QueryBuilder__20, String, QueryBuilder.QueryBuilder__19>
+          addAggregation,
+      SelectedField selectedField,
+      SupportedAggregationFunction supportedAggregationFunction) {
+    List<String> args = getAndValidateArgs(arguments, supportedAggregationFunction);
+    String column = getAndValidateColumn(args, supportedAggregationFunction);
+
+    String alias = selectedField.getAlias();
+    QueryBuilder.QueryBuilder__19 withAggregation = addAggregation.apply(queryBuilder, column);
+    if (alias != null) {
+      withAggregation.as(alias);
+    }
+  }
+
+  private String getAndValidateColumn(
+      List<String> args, SupportedAggregationFunction supportedAggregationFunction) {
+    String column = dbColumnGetter.getDBColumnName(table, args.get(0));
+    if (column == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The column name: %s provided for the %s function does not exists.",
+              args.get(0), supportedAggregationFunction.getName()));
+    }
+    return column;
+  }
+
+  private List<String> getAndValidateArgs(
+      Map<String, Object> arguments, SupportedAggregationFunction supportedAggregationFunction) {
+    Object argsToValidate = arguments.get("args");
+    if (argsToValidate == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The args provided for a function: %s are null, but it is not allowed.",
+              supportedAggregationFunction.getName()));
+    }
```

Comment:
Maybe remove this check if you apply my other suggestion of making it non-null in the schema.